### ISSUE TITLE
Chatglm2-6b 不能直接传入 attention_mask

### ISSUE THREAD

**Issue by Haskely** [yangjianxin1/Firefly#91]
根据 https://huggingface.co/THUDM/chatglm2-6b/blob/v1.0/modeling_chatglm.py#L191 ，模型 forward 若传入了 attention_mask 不会自动转换为 causal attention mask，会导致模型训练时 forward 能看到全部 tokens。

但是 https://github.com/yangjianxin1/Firefly/blob/master/component/dataset.py#L96 传入了全 1 的 mask

**Comment by Elenore1997** [comment]
同样对该代码的attention mask的构造有疑问，这样子微调不是看到左右两边的信息了么？

**Comment by mvllwong** [comment]
你这不就是我说的吗，直接在我那讨论嘛

**Comment by mvllwong** [comment]
> 同样对该代码的attention mask的构造有疑问，这样子微调不是看到左右两边的信息了么？

其他isssue里面一次回答的结果直接调成多次问答，可以看看之前的包括我的讨论

**Comment by mvllwong** [comment]
这个问题不仅如此，事实上gllm自制的attention mask维度和firefly自以为的维度是不同的，甚至和prefix encode mask叠加时候都可能有问题

**Comment by Haskely** [comment]
## 更新

chatglm2 在这里添加了 causal attention mask：https://huggingface.co/THUDM/chatglm2-6b/blob/0ade0d38ac00258ae09450696315c2ff0b1faf12/modeling_chatglm.py#L627

所以不存在该问题。
